framework-cloud/framework-cloud-api/src/main/java/org/kie/cloud/api/DeploymentScenarioBuilderFactoryLoader.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.api;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+public class DeploymentScenarioBuilderFactoryLoader {
+    public static DeploymentScenarioBuilderFactory getInstance() {
+        List<DeploymentScenarioBuilderFactory> deploymentScenarioBuilderFactories = getFactories();
+        if (deploymentScenarioBuilderFactories.size() == 1) {
+            return deploymentScenarioBuilderFactories.get(0);
+        } else if (deploymentScenarioBuilderFactories.size() == 0) {
+            throw new RuntimeException("No cloud API implementation was found");
+        } else {
+            StringBuilder cloudImplementations = new StringBuilder();
+            deploymentScenarioBuilderFactories.stream().forEach(x -> cloudImplementations.append(x.getCloudEnvironmentName() + ", "));
+
+            throw new RuntimeException("Multiple cloud API implementations detected - please select one from: " + cloudImplementations.toString());
+        }
+    }
+
+    public static DeploymentScenarioBuilderFactory getInstance(String environmentName) {
+        List<DeploymentScenarioBuilderFactory> deploymentScenarioBuilderFactories = getFactories();
+        Map<String, DeploymentScenarioBuilderFactory> factoryMap = new HashMap<>();
```

Comment:
Thank you for nice stream. But I'm not sure about this. I don't like huge streams which replace whole methods.